### DISCUSSION TITLE
Authentication via API call

### DISCUSSION THREAD

**Discussion by lniklison** [jasontaylordev/CleanArchitecture#830] [Q&A]
I've been looking around the project for a way to log in to the application via postman instead of the UI project. Is there an endpoint I can use to get a token or something? I've already searched the requests that the front end sends out but couldn't figure out anything.

 I was thinking about creating an authentication controller that consumes the Authentication service in Infrastructure/Identity but wanted to make sure that there isn't something already implemented.

**Comment by RainerKoller** [comment]
Hi
I am looking for the same functionallity. Login via API (i.e. Postman) and get a JWT token that way. Not via UI. Do you, in the mean time, have a proposal for this?
Very keen to get an answer.

Thanks in advance

- Rainer

**Reply by jasontaylordev** [reply]
You can learn more here; https://learn.microsoft.com/en-us/aspnet/core/security/authentication/identity-api-authorization?view=aspnetcore-7.0.

**Reply by spa5k** [reply]
@jasontaylordev this seems to be same as what is used in current Architecture and not through APIs.

Also, @lniklison @RainerKoller  were you able to do it? If possible, can you share the code.

**Comment by Neptune-Apps** [comment]
This is based on the latest code, so might be different from March time.

The User API endpoints do not require authorization. So from Postman, after setting up a user (you can call /api/users/register and then update the EmailConfirmed column manually in DB if you have not got email set up), you would then call 
/api/users/login which provides the access token.

In Postman we then use the Tests tab and a global variable ({{access_token}}) to use that access token for the other api calls that require authorization

```
const responseJson = pm.response.json();
pm.collectionVariables.set('access_token', responseJson.access_token);
```

Hope that makes sense and helps.
